File: app/src/main/java/com/codemybrainsout/imageviewer/view/detail/DetailContract.java
package com.codemybrainsout.imageviewer.view.detail;

import android.content.Context;
import android.graphics.Bitmap;
import android.net.Uri;

import com.codemybrainsout.imageviewer.model.Exif;
import com.codemybrainsout.imageviewer.model.Photo;
import com.codemybrainsout.imageviewer.model.User;
import com.codemybrainsout.imageviewer.view.base.BasePresenter;
import com.codemybrainsout.imageviewer.view.base.BaseView;

import retrofit2.http.Url;

/**
 * Created by ahulr on 09-06-2017.
 */

public class DetailContract {

    interface Presenter extends BasePresenter {

        void loadPhoto();

        void downloadPhoto(Context context, String url);

        void setAsWallpaper(Context context, String url);

        void showExif(Exif exif);

        void openUserDetails(User user);
    }

    interface View extends BaseView<Presenter> {

        void showPhoto(Photo photo);

        void showExif(Exif exif);

        void showUserDetails(User user);
    }

}

File: app/src/main/java/com/codemybrainsout/imageviewer/view/detail/DetailContract.kt
package com.codemybrainsout.imageviewer.view.detail

import com.codemybrainsout.imageviewer.view.base.BasePresenter
import com.codemybrainsout.imageviewer.view.base.BaseView
import android.content.Context
import com.codemybrainsout.imageviewer.model.*

/**
 * Created by ahulr on 09-06-2017.
 */
class DetailContract {

    interface Presenter : BasePresenter {
        fun loadPhoto(photoId: String)
        fun downloadPhoto(context: Context, url: String?)
        fun setAsWallpaper(context: Context, url: String?)
        fun showExif(exif: Exif?)
        fun openUserDetails(user: User?)
    }

    interface View : BaseView<Presenter?> {
        fun showPhoto(photo: Photo)
        fun showExif(exif: Exif?)
        fun showUserDetails(user: User?)
    }
}